modules/vye/app/models/concerns/vye/needs_enrollment_verification.rb

```diff
@@ -148,33 +165,38 @@ def eval_case1a
         payment_date: @award.payment_date,
         trace: :case1a
       )
+
+      true
     end
 
     def eval_case1b
-      return unless dlc_before_ldpm?
+      return unless dlc_before_ldpm? || date_last_certified.blank?
       return unless @award.award_ind_current?
       return if @award.award_end_date.blank?
       return if are_dates_the_same(@award.award_end_date, date_last_certified)
-      return unless current_rec_ended
+      return unless current_rec_ended?
+      return if ldpm_before_aed? && are_dates_the_same(@award.award_begin_date, @award.award_end_date)
 
       push_enrollment(
         award_id: @award.id,
         act_begin: date_last_certified,
-        act_end: @award.award_end_date - 1.day,
+        act_end: aed_minus1,
         number_hours: @award.number_hours,
         monthly_rate: @award.monthly_rate,
         payment_date: @award.payment_date,
         trace: :case1b
       )
+
+      true
     end
 
     def eval_case2
```

Comment:
Is it possible to give these numbered cases more descriptive names? They all seem to be very similar and I'm having a hard time understanding what makes them different.

Reply:
@rmtolmach It's hard to describe them other than as "Case". We refer to this as the Enrollment Logic as a whole and each "Case" is a check that is completed to determine if the veteran has any verifications that are ready to be verified in MGIB.   In short, this logic runs through key pieces of data such as the award's start date, end date, and date last certified for that individual veteran and determines if the current data supplied is ready to be verified. There are several different times a veteran's enrollment verification is ready to be verified and each "Case" is one of those reasons. Please let me know if you would like to meet up and talk this through.

Reply:
Having an understanding of the code isn't just for ME, it's a maintainability issue. Contracts don't stay around forever, so I want to make sure the team maintaining this code down the road understands what's happening in this file. There's no way these case names could be more descriptive than numbering them? Is there something that makes each case unique? If they need to have the numbering convention, could a comment be added to each method? For example, the comments in this file explaining the acronyms are helpful:
```ruby
   # date_last_certified is before award_begin_date
    def dlc_before_abd?
```